addons/product/views/product_views.xml

```diff
@@ -125,12 +125,16 @@
                                     </div>
                                     <label
                                         for="standard_price"
-                                        invisible="(product_variant_count &gt; 1 and not is_product_variant) or type == 'combo'"
+                                        invisible="((product_variant_count &gt; 1
+                                                    or (product_variant_count == 0 and is_dynamically_created))
+                                                    and not is_product_variant) or type == 'combo'"
```

Comment:
I would reorder the attributes so that they are clearer to understand and read:
```suggestion
                                        invisible="type == 'combo' 
                                                    or (not is_product_variant and (
                                                         product_variant_count &gt; 1
                                                         or (product_variant_count == 0 and is_dynamically_created)
                                                     ))
```